packages/compile-service/src/service/zk-compile.ts

```diff
@@ -32,26 +33,42 @@ export class ZkCompileService {
   async compileAndCreateDeployUnsignTx(
     payerAddress: string,
     zkDbPrivateKey: PrivateKey,
-    merkleHeight: number
+    merkleHeight: number,
+    databaseName: string
   ): Promise<UnsignedTransaction> {
-    const zkDbPublicKey = PublicKey.fromPrivateKey(zkDbPrivateKey);
-
-    const start = performance.now();
+    try {
+      const zkDbPublicKey = PublicKey.fromPrivateKey(zkDbPrivateKey);
 
-    const zkWrapper = await this.getSmartContract(merkleHeight);
+      const start = performance.now();
 
-    let unsignedTx = await zkWrapper.createAndProveDeployTransaction({
-      sender: PublicKey.fromBase58(payerAddress),
-      zkApp: zkDbPublicKey,
-    });
+      const zkWrapper = await this.getSmartContract(merkleHeight);
 
-    unsignedTx = unsignedTx.sign([zkDbPrivateKey]);
-    const end = performance.now();
-    logger.info(
-      `Deploy ${zkDbPublicKey.toBase58()} take ${(end - start) / 1000}s`
-    );
-
-    return unsignedTx.toJSON();
+      let unsignedTx = await zkWrapper.createAndProveDeployTransaction({
+        sender: PublicKey.fromBase58(payerAddress),
+        zkApp: zkDbPublicKey,
+      });
+
+      unsignedTx = unsignedTx.sign([zkDbPrivateKey]);
+
+      await ModelDbSetting.getInstance().updateSetting(databaseName, {
+        appPublicKey: zkDbPublicKey.toBase58(),
+        deployStatus: "ready",
+      });
+
+      const end = performance.now();
+      logger.info(
+        `Deploy ${zkDbPublicKey.toBase58()} take ${(end - start) / 1000}s`
+      );
+
+      return unsignedTx.toJSON();
+    } catch (error) {
+      logger.error(`Cannot compile & deploy: ${databaseName}`, logger);
+      await ModelDbSetting.getInstance().updateSetting(databaseName, {
+        appPublicKey: undefined,
+        deployStatus: "failed",
```

Comment:
I think, here we should have 2 states: `deployed` | `not_deployed`. Maybe we should store the transaction regardless of success/failed and store the failed status there alongside with errorMessage

Reply:
resolved